src/pages/Signup/components/FirstSignupForm/SignupInput/Input.tsx

```diff
@@ -0,0 +1,81 @@
+import { ChangeEvent, useState } from "react"
+import * as S from "./Input.Styles"
+import type { AllowedInputType } from "../../../types/index"
+import VisibilityIcon from "@mui/icons-material/Visibility"
+import VisibilityOffIcon from "@mui/icons-material/VisibilityOff"
+
+type OnChange = ({ target }: ChangeEvent<HTMLInputElement>) => void
+
+interface InputProp {
+  type: AllowedInputType
+  placeholder: string
+  name: string
+  value: string
+  onChange?: OnChange
+}
+
+type InputTypeList = {
+  [key in AllowedInputType]: string
+}
+
+const Input = ({ type, name, value, placeholder, onChange }: InputProp) => {
+  const [isVisible, setIsVisible] = useState(false)
+
+  const inputTypeList: InputTypeList = {
+    email: "text",
+    nickname: "text",
+    password: "password",
+  }
+
+  const handleVisible = () => {
+    setIsVisible(!isVisible)
+  }
+
+  const getInputFieldByType = (type: AllowedInputType) => {
+    switch (type) {
+      case "email":
+      case "nickname":
+        return (
+          <S.Input
+            type={inputTypeList[type]}
```

Comment:
case로 나누어줘서 type이 text인지 password인지 정해져있으니까, 값을 바로 줘도 괜찮을 것 같은데 어떻게 생각하시나요~?
```suggestion
            type="text"
```

Reply:
type으로 email을 넣으면 기본적인 이메일에 대한 validation이 나옵니다!
기본적으로 나오는 validation이 별로라고 해서 email 인 경우에는 type을 text로 넣기 위해서 이렇게 했었습니다! 👍
그런데 지금 생각해보니 case로 나누어줘서 그냥 type에 'text'라고 바로 넣어도 괜찮겠네요 ㅎㅎ
좋은 의견 감사합니다! 😊